### PR TITLE
Output EDR by default

### LINKED ISSUE #70: Add EDR/(Compressed XVG) switch for dH data output
Add an option in the `runinput.yml` file for the user to specify how they would like energy data to be written out. The options should be between zipped `xvg` files and an `edr` file

### PR DESCRIPTION
**Pull request by ianmkenney**
Implement behavior such that EDR files are produced during FEP window runs by default. 

FIxes #70.

### CODE REVIEW

File: mdpow/fep.py [hunk 1]
```diff
@@ -621,6 +621,7 @@ def _setup(self, component, lmbda, foreign_lambdas=None, **kwargs):
 
         wdir = self.wdir(component, lmbda)
         kwargs.setdefault('couple-intramol', 'no')
+        kwargs.setdefault('e','')
```
Comment (orbeckst) [inline mdpow/fep.py]: PEP8 style:
```python
kwargs.setdefault('e', '')
```
(space after commas)

**Comment by orbeckst** [comment]
@ianmkenney typically you assign a PR to someone else for review and merge. I added myself so that I get emails when anything changes. When you're done, remove the WIP tag and also ping me with a "please review @orbeckst" or similar.

File: mdpow/templates/bar_opls.mdp [hunk 2]
```diff
@@ -110,6 +110,7 @@ lincs-iter               = 1
 
 ; Free energy control stuff
 free_energy              = yes
+separate-dhdl-file       = no
```
Comment (orbeckst) [inline mdpow/templates/bar_opls.mdp]: You need to also set at least the option to compute *all* foreign lambdas.

**Comment by ianmkenney** [comment]
Should we include issue #76 in this PR?

**Comment by orbeckst** [comment]
Yes, makes sense with "added the MBAR method": the MDP file should be compatible with the advertised capabilities.

**Comment by ianmkenney** [comment]
@orbeckst please review

File: mdpow/templates/bar_opls.mdp (hunk 2 above)
Comment (ianmkenney) [inline mdpow/templates/bar_opls.mdp]: Added a test that checks that the edr is produced. I check the mdps produced and the contents of the edr manually with `gmx dump -e md.edr`; it all seems correct to me

File: mdpow/fep.py [hunk 3]
```diff
@@ -583,20 +584,9 @@ def setup(self, **kwargs):
...
-                xlambdas = [None] + list(lambdas) + [None]
-                for l in lambdas:
-                    foreign_lambdas = [xlambdas[xlambdas.index(l) - 1],
-                                       xlambdas[xlambdas.index(l) + 1]]
-                    # set up gromacs job for each FEP window in BAR
-                    params = self._setup(component, l,
-                                         foreign_lambdas=foreign_lambdas, **kwargs)
-            else:
-                raise ValueError("Unknown method {}".format(self.method))
+            for l in lambdas:
```
Comment (orbeckst) [inline mdpow/fep.py]: This works for TI, BAR, and MBAR?

Don't you have to add the end points or is it really sufficient just to repeat all lambdas?

File: mdpow/fep.py [hunk 4]
```diff
@@ -625,46 +615,31 @@ def _setup(self, component, lmbda, foreign_lambdas=None, **kwargs):
...
+                      mainselection=None,
+                      runtime=self.runtime,
+                      ref_t=self.Temperature,    # TODO: maybe not working yet, check _setup()
+                      gen_temp=self.Temperature, # needed until gromacs.setup() is smarter
+                      qname=self.tasklabel(component,lmbda),
+                      free_energy='yes',
+                      couple_moltype=self.molecule,
+                      init_lambda_state=lambda_index,
+                      fep_lambdas=foreign_lambdas,
+                      calc_lambda_neighbors=-1,
```
Comment (orbeckst) [inline mdpow/fep.py]: Does `calc_lambda_neighbors=-1` work for Gromacs 4.6.x?

If not, you need code for Gromacs 5.x/2016 and Gromacs 4.6.x

File: mdpow/templates/bar_opls.mdp [hunk 5]
```diff
@@ -110,10 +110,12 @@ lincs-iter               = 1
...
 ;  init for bar
-init_lambda              = 0.0
+init_lambda_state        = 0.0
```
Comment (orbeckst) [inline mdpow/templates/bar_opls.mdp]: Is this valid for Gromacs 4.6.x?

File: mdpow/fep.py (hunk 4 above)
Comment (ianmkenney) [inline mdpow/fep.py]: Fortunately, all of the changes are supported by gromacs 4.6.x docs.

**Comment by orbeckst** [comment]
Great. 

What will tests look like?

--
Oliver Beckstein
email: orbeckst@gmail.com

File: mdpow/fep.py (hunk 3 above)
Comment (ianmkenney) [inline mdpow/fep.py]: It's should be sufficient to supply the full lambda vector along with the `init_lambda_state`. The iterations over lambda was originally in the code; lambdas would be stripped out of the lambda list and the remaining lambdas would be put into the `fep-lambdas` mdp option. Now each MDP will contain information on the full lambda vector, which outputs information more consistently.

**Comment by ianmkenney** [comment]
I'm not sure what you mean by what the tests look like. Tests regarding what functionality?

File: mdpow/templates/bar_opls.mdp (hunk 5 above)
Comment (ianmkenney) [inline mdpow/templates/bar_opls.mdp]: yes

Comment (orbeckst) [inline mdpow/templates/bar_opls.mdp]: nice

**Comment by orbeckst** [comment]
tests: How will we know that these changes do what they are supposed to do? Do we get correct answers for
- MBAR
- TI
- (BAR)

`test_fep.py` only tests the schedules but not the actual calculation so we have no idea what these changes do...

Basically, I will merge the changes but with the understanding that we need tests one way or another. You can use your local benzene runs for comparison for a start or run MDPOW from the commit *before this merge/rebase* but you need to make sure that you truely compare old vs new.

**Comment by orbeckst** [comment]
Github rebase failed. I'll do it manually.

**Comment by orbeckst** [comment]
Rebased against develop and `git push -f`. Waiting for travis.

**Pull request merged**